plugins/auth-backend/src/lib/resolvers/CatalogAuthResolverContext.ts

```diff
@@ -123,8 +123,22 @@ export class CatalogAuthResolverContext implements AuthResolverContext {
       const res = await this.catalogApi.getEntities({ filter }, { token });
       result = res.items;
     } else if ('filter' in query) {
+      const filter = [query.filter].flat().map(value => {
+        if (
+          !(
+            'kind' in
```

Comment:
oh wait, after adding this i realize - doing `'kind' in <array>` makes no sense at all, you need to do `<array>.includes('kind')` or `Object.keys(value).some(k => k.toLocaleLowerCase('en-US') === 'kind')`

Reply:
Sorry for that, it's fixed now and I have added a test. Please let me know if I can improve anything about the test setup!